contracts/scripts/standard-bridge/DeployStandardBridge.s.sol

```diff
@@ -13,75 +13,145 @@ import {Strings} from "@openzeppelin/contracts/utils/Strings.sol";
+        require(settlementFinalizationFee != 0, SettlementFinalizationFeeNotSet(settlementFinalizationFee));
+    }
+}
+
+contract DeploySettlementGateway is BridgeBase {
+    error DeployerMustHaveGenesisAllocation(uint256 balance, uint256 expected);
+    error FailedToFundAllocator(address allocator);
+
     function run() external {
 
         vm.startBroadcast();
 
-        address relayerAddr = vm.envAddress("RELAYER_ADDR");
-        require(relayerAddr != address(0), "RELAYER_ADDR is not set");
-        address whitelistAddr = vm.envAddress("WHITELIST_ADDR");
-        require(whitelistAddr != address(0), "WHITELIST_ADDR is not set");
+        address relayerAddr = getRelayerAddress();
+        uint256 l1FinalizationFee = getL1FinalizationFee();
+        uint256 settlementFinalizationFee = getSettlementFinalizationFee();
+
+        require(address(msg.sender).balance >= DEPLOYER_GENESIS_ALLOCATION, 
+            DeployerMustHaveGenesisAllocation(address(msg.sender).balance, DEPLOYER_GENESIS_ALLOCATION));
+
+        address allocatorProxy = Upgrades.deployUUPSProxy(
+            "Allocator.sol",
+            abi.encodeCall(Allocator.initialize,
+            (msg.sender)) // Owner
+        );
+        Allocator allocator = Allocator(payable(allocatorProxy));
+        console.log("Allocator deployed to:", address(allocator));
+
+        (bool success, ) = payable(address(allocator)).call{value: DEPLOYER_GENESIS_ALLOCATION - MEV_COMMIT_CHAIN_SETUP_COST}("");
+        require(success, FailedToFundAllocator(address(allocator)));
 
         address sgProxy = Upgrades.deployUUPSProxy(
             "SettlementGateway.sol",
             abi.encodeCall(SettlementGateway.initialize,
-            (whitelistAddr,
+            (allocatorProxy,
             msg.sender, // Owner
             relayerAddr,
-            1, 1)) // Fees set to 1 wei for now
+            settlementFinalizationFee, // SettlementGateway._finalizationFee
+            l1FinalizationFee)) // SettlementGateway._counterpartyFee
         );
-        SettlementGateway gateway = SettlementGateway(payable(sgProxy));
+        SettlementGateway settlementGateway = SettlementGateway(payable(sgProxy));
         console.log("Standard bridge gateway for settlement chain deployed to:",
-            address(gateway));
-        address allocatorProxy = Upgrades.deployUUPSProxy(
-            "Allocator.sol",
-            abi.encodeCall(Allocator.initialize, (msg.sender))
-        );
-        Allocator allocator = Allocator(payable(allocatorProxy));
-        console.log("Allocator deployed to:", address(allocator));
+            address(settlementGateway));
 
-        allocator.addToWhitelist(address(gateway));
-        console.log("Settlement gateway has been whitelisted. Gateway contract address:", address(gateway));
+        allocator.addToWhitelist(address(settlementGateway));
+        console.log("Settlement gateway has been whitelisted");
 
         string memory jsonOutput = string.concat(
```

Comment:
nit: maye consider using something like the following: https://github.com/bmeredith/solidity-json-writer

Reply:
Nice this makes sense, but we'd seemingly have to make it compatible with `forge-std/console.sol` output. That repo is hardhat based